src/util/generate/generate-dir-map.ts

```diff
@@ -0,0 +1,64 @@
+import fs from 'fs'
+import { promisify } from 'util'
+import { join, parse } from 'path'
+import { allDetectors } from './detect-from-manifests'
+import ignore from '../ignored'
+
+const ignored = ignore.split('\n')
+const readdir = promisify(fs.readdir)
+const stat = promisify(fs.stat)
+
+export type dirMap = {
+  absolute: string,
+  dir: {
+    [key: string]: dirMap
+  }
+  extensions: {
+    [type: string]: number | string
+  },
+  manifests: string[]
+}
+
+export async function generateDirMap(dir: string, ignore: { [key: string]: true }, rootDir: string = dir): Promise<dirMap> {
+  const result = await readdir(dir)
+  const map: dirMap = {
+    absolute: dir,
+    dir: {},
+    extensions: {},
+    manifests: []
+  }
+
+  await Promise.all(result.map(async (part: string) => {
+    const absolutePath = join(dir, part)
+    const pathStat = await stat(absolutePath)
+    if (['.nowignore', 'now.json'].includes(part)) return
+
+    if (ignored.includes(part)) {
+      ignore[part] = true
```

Comment:
I don't think you need to set `ignore[part] = true` here because default ignores don't need to be added to the `.nowignore` file, they're implicit.

Reply:
It's so they get added explicitly, so the user knows what's being ignored out of the defaults